Datasnap/Classes/Event Properties/Identifier.m

```diff
@@ -46,8 +47,25 @@ - (NSDictionary*)convertToDictionary
         @"mobile_device_google_advertising_id" : self.mobileDeviceGoogleAdvertisingId ? self.mobileDeviceGoogleAdvertisingId : [NSNull null],
         @"mobile_device_google_advertising_id_opt_in" : self.mobileDeviceGoogleAdvertisingIdOptIn ? self.mobileDeviceGoogleAdvertisingIdOptIn : [NSNull null],
         @"unknown" : self.unknown ? self.unknown : [NSNull null],
-        @"facebook_uuid" : self.facebookUuid ? self.facebookUuid : [NSNull null]
+        @"facebook_uuid" : self.facebookUuid ? self.facebookUuid : [NSNull null],
+        @"hashed_email" : [self sha1:self.hashedEmail]
     };
     return dictionary;
 }
+- (NSString*)sha1:(NSString*)input
```

Comment:
This method appears here and in Datasnap.m.  I'm not sure it belongs in either location.  This seems like a string helper method...  You could probably extract this into a category - something like NSString+Helpers.  This would dry up the code a lot, esp if you end up needing it beyond these two areas.  I would also rename it to further clarify the purpose even if it's just `toSha1`.  From there it's a simple `NSString *converted = [myStringVariable toSha1];`